src/node.ts

```diff
@@ -0,0 +1,190 @@
+import { RangeResolver } from "./resolver";
+
+export type ReferencedValue = { dataPointer: MemoryPointer; value: Buffer };
+export type MemoryPointer = { offset: number; length: number };
+
+export class BPTreeNode {
+	public dataHandler: RangeResolver;
+	public pointers: MemoryPointer[];
+	public keys: ReferencedValue[];
+
+	constructor(
+		dataHandler: RangeResolver,
+		pointers: MemoryPointer[],
+		keys: ReferencedValue[]
+	) {
+		this.dataHandler = dataHandler;
+		this.pointers = pointers;
+		this.keys = keys;
+	}
+
+	leaf(): boolean {
+		return this.pointers.length === this.keys.length;
+	}
+
+	async readFrom(): Promise<number> {
```

Comment:
This function is probably better off `static`. In practice, the `readFrom()` pattern is also a golang pattern, so something like

```
static async fromMemoryPointer(mp: MemoryPointer, resolver: Resolver): Promise<BPTreeNode> {
```

It's fine to leave for now and address in the next PR.